### DISCUSSION TITLE
STM32F407 mit EVE3-50G-TPC von MatrixOrbital

### DISCUSSION THREAD

**Discussion by fragefreund** [RudolphRiedel/FT800-FT813#109] [Q&A]
Ich bin relativ neu beim Thema Display mit Mikrocontroller.
Ich hab mich für den STM32F407 DISCOVERY entschieden und würde den gerne mit dem EVE3-50G-TPC von MatrixOrbital verbinden. ich hab es versucht dies mithilfe der bereitgestellten library alles zum Laufen zu bekommen, dass hat aber noch nicht geklappt.

Ich hab die Headerdateien: EVE.h, EVE_commands.h, EVE_config.h, EVE_supplemental.h und EVE_target.h zu meinem Projekt inzugefügt und ein entsprechendes define zur Auswahl meines Displays in der main.c-Datei gemacht. Leider scheint dies vom Code nicht erkannt zu werden.

Des Weiteren hab ich die C-Dateien: EVE_commands.c, EVE_supplemental.c und EVE_target.c eingebunden.
Als SPI hab ich in meiner IDE den SPI1 ausgewählt und CS ist der Pin PB12 mit dem User-label: EVE_CS (siehe Bilder).
Ich wäre froh, wenn mir jemand helfen könnte. 
Vielen Dank schon mal :)

![EVE_EVE3_50G](https://github.com/RudolphRiedel/FT800-FT813/assets/159798566/1b6243ad-0ea0-4963-baf7-939b6e19894e)
![SPI_Einstellungen1](https://github.com/RudolphRiedel/FT800-FT813/assets/159798566/b2fbcb4e-8b99-481a-87f3-546600df0996)
![SPI_Einstellungen2](https://github.com/RudolphRiedel/FT800-FT813/assets/159798566/29b3d6f8-0f51-44a6-a9be-a7f600b786bc)

**Comment by RudolphRiedel** [comment]
Hallo,

zufällig habe ich das vor kurzem erst hier beschrieben, allerdings für einen STM32 der noch gar nicht unterstützt wurde:
http://www.brtcommunity.com/index.php?topic=495.0

Der F4 wird erkannt mit dem Makro "STM32F4".
Und einzustellen sind die Makros in den Projekt-Einstellungen:
![grafik](https://github.com/RudolphRiedel/FT800-FT813/assets/31180093/24c26cd8-194e-498f-bb49-ecb5f62eef9f)

Die EVE_target.h alleine reicht nicht, für den STM32 braucht es noch die EVE_target/EVE_target_STM32.h.
Da stehen auch die Pins und der SPI drin.

Dann muss man auch noch schauen, was überhaupt in der MX_SPI1_Init() drin steht.
Wenn da kein LL_SPI_Enable() am Ende ist, dann ist der SPI auch nicht aktiv.
Und ich hatte das schon das in der Init die Konfiguration der SPI Pins gefehlt hat.

Den Takt beschreibt das ja auch nur begrenzt.

Als nächsten Schritt vor dem Anschließen des Displays würde ich schauen ob was aus dem SPI raus kommt und bei welchem Takt.

**Comment by fragefreund** [comment]
Hallo, 

vielen Dank für die schnelle Antwort. Ich bin der Anleitung im erwähnten Beitrag gefolgt und das Programm lässt sich kompilieren. Der SPI sendet mit der eingestellten Frequenz von 2 MHz, die Verbindung zu den Pins cs und pdn funktioniert ebenfalls.

Nun wollte ich wie im Forum erwähnt das Beispiel: "EVE_Test_SAMC21_EVE2-50G" dort kam es dann zu den unten angehängten Fehlern.

Wie werde ich die denn los oder ist es sinnvoller am Anfang mir zum Beispiel nur einen Text anzeigen zu lassen oder ähnliches?
![Fehlermeldungen](https://github.com/RudolphRiedel/FT800-FT813/assets/159798566/e3546c7a-8f51-4009-88bb-a7839d051d0f)

**Reply by RudolphRrr** [reply]
Na, die Main,c aus dem SAMC21 Beispiel kann nicht direkt funktionieren, die zu übernehmen habe ich auch nicht geschrieben,
Die tft.c und tft_data,c sind im Grunde für alle Beispiele identisch, nur ist die Ordner-Struktur bei den PlatformIO Beispielen anders als bei den Beispielen für das Microchip-Studio.
Und was ich im Forum für die STM32CubeIDE beschrieben habe ist das gleiche was ich im Microchip Studio gemacht habe,
ich habe ein Unterverzeichnis "EmbeddedVideoEngine" angelegt und da die notwendigen Dateien rein geworfen.

In der tft.c für den SAMC21 steht entsprechend:
#include "EmbeddedVideoEngine/EVE.h"
In der tft.c für ESP32 mit PlatformIO steht:
#include "EVE.h"

In dem Forums-Beitrag habe ich auch beschrieben wie ich die main.c ändern würde die STM32CubeIDE generiert hat:

```
/* USER CODE BEGIN Includes */
#include "tft.h"

/* USER CODE END Includes */

...

int main(void)
{
...
  /* Initialize all configured peripherals */
  MX_GPIO_Init();
  MX_SPI1_Init();

  /* USER CODE BEGIN 2 */
  TFT_init();

  TFT_display();

  /* USER CODE END 2 */

  /* Infinite loop */
  /* USER CODE BEGIN WHILE */
  while (1)
  {
    /* USER CODE END WHILE */

    /* USER CODE BEGIN 3 */
  }
  /* USER CODE END 3 */
}
```

Das gibt aber nur ein statisches Bild, das ist ein erster Test.

Das hier ist mehr für den STM32, also die Hauptschleife:
https://github.com/RudolphRiedel/FT800-FT813/blob/5.x/examples/EVE_Test_STM32_RiTFT50_PlatformIO/src/main.c

Nur, so richtig sauber ist das nicht, das ganze Beispiel ist nicht funktional, das ist mehr ein Test ob das compiliert.
